File: algorithms/1603.design-parking-system/solution.go
type ParkingSystem struct {
	parkingSpaces []int
}


func Constructor(big int, medium int, small int) ParkingSystem {
	return ParkingSystem { []int{ 0, big, medium, small } }
}


func (this *ParkingSystem) AddCar(carType int) bool {
	if this.parkingSpaces[carType] == 0 {
		return false
	}
	this.parkingSpaces[carType] -= 1
	return true
}


/**
 * Your ParkingSystem object will be instantiated and called as such:
 * obj := Constructor(big, medium, small);
 * param_1 := obj.AddCar(carType);
 */

File: algorithms/1603.design-parking-system/solution.ts
class ParkingSystem {
  parkingSpaces = [0, 0, 0];

  constructor(big: number, medium: number, small: number) {
    this.parkingSpaces = [big, medium, small];
  }

  addCar(carType: number): boolean {
    if (this.parkingSpaces[carType - 1] > 0) {
      this.parkingSpaces[carType - 1]--;
      return true;
    } else {
      return false;
    }
  }
}

/**
 * Your ParkingSystem object will be instantiated and called as such:
 * var obj = new ParkingSystem(big, medium, small)
 * var param_1 = obj.addCar(carType)
 */
